src/io/flutter/FlutterUtils.java

```diff
@@ -278,129 +307,65 @@ public static PluginId getPluginId() {
   }
 
   /**
-   * Returns true if passed pubspec declares a flutter or flutter_web dependency.
-   * <p>
-   * This method is provided explicitly instead of calling declaresFlutter(VirtualFile) || declaresFlutterWeb(VirtualFile) so only one read
-   * and parsing of the pubspec is made.
+   * Returns a structured object with information about the Flutter properties of the given
+   * pubspec file.
    */
-  public static boolean declaresFlutterAny(@NotNull final VirtualFile pubspec) {
-    // It uses Flutter if it contains:
-    // dependencies:
-    //   flutter:
+  public static FlutterPubspecInfo getFlutterPubspecInfo(@NotNull final VirtualFile pubspec) {
+    // It uses Flutter if it contains 'dependencies: flutter'.
+    // It uses Flutter if it contains 'dependencies: flutter_web'.
+    // It's a plugin if it contains 'flutter: plugin'.
+
+    final FlutterPubspecInfo info = new FlutterPubspecInfo(pubspec.getModificationStamp());
 
     try {
       final Map<String, Object> yamlMap = readPubspecFileToMap(pubspec);
-      if (yamlMap == null) {
-        return false;
-      }
+      if (yamlMap != null) {
+        // Special case the 'flutter' package itself - this allows us to run their unit tests from IntelliJ.
+        final Object packageName = yamlMap.get("name");
+        if ("flutter".equals(packageName)) {
+          info.flutter = true;
+        }
 
-      // Special case the 'flutter' package itself - this allows us to run their unit tests from IntelliJ.
-      final Object name = yamlMap.get("name");
-      if ("flutter".equals(name)) {
-        return true;
-      }
+        // Check the dependencies.
+        final Object dependencies = yamlMap.get("dependencies");
+        if (dependencies instanceof Map) {
+          info.flutter |= ((Map)dependencies).containsKey("flutter");
+          info.flutterWeb = ((Map)dependencies).containsKey("flutter_web");
```

Comment:
why does only one of these use `|=`

Reply:
The `flutter` field can sometimes be set to true above; I'll add a comment about that.